SmartTutor/ContentModel/LearningObjects/Challenge.cs

```diff
@@ -0,0 +1,47 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using SmartTutor.ContentModel.LearningObjects.MetricRules;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SmartTutor.ContentModel.LearningObjects
+{
+    public class Challenge : LearningObject
+    {
+        public string Url { get; internal set; }
+        public List<CaDETClass> ResolvedClasses { get; internal set; }
+        public List<MetricRangeRule> ClassMetricRules { get; internal set; }
+        public List<MetricRangeRule> MethodMetricRules { get; internal set; }
+
+        // TODO: This is basic strategy, implement strategy pattern interface 
+        public bool CheckSubmittedChallengeCompletion(List<CaDETClass> submittetClasses)
+        {
+            List<CaDETMember> submittedMethods = GetMethodsFromClasses(submittetClasses);
+            if (!CheckClassesCompletion(submittetClasses) || !CheckMethodsCompletion(submittedMethods))
+                return false;
+            return true;
+        }
+
+        private List<CaDETMember> GetMethodsFromClasses(List<CaDETClass> caDETClasses)
+        {
+            return caDETClasses.SelectMany(c => c.Members.Where(m => m.Type.Equals(CaDETMemberType.Method))).ToList();
+        }
+
+        private bool CheckClassesCompletion(List<CaDETClass> caDETClasses)
```

Comment:
I'm not a fan of the names CheckClassesCompletion and CheckMethodsCompletion.

Maybe "ValidateClassMetricRules"?